framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -936,6 +953,19 @@ public AnnotationMirror createArrayLenRangeAnnotation(Range range) {
         return createArrayLenAnnotation(ValueCheckerUtils.getLengthsForStringValues(values));
     }
 
+    /**
+     * Converts an {@code StringVal} annotation to an {@code MatchesRegex} annotation that matches
+     * exactly the string values listed in the {@code StringVal}.
```

Comment:
This code does not do that.  To achieve this, you would need to call `Pattern.quote` and also add `^` and `$` anchors.

Also note that the strings are not necessarily legal regular expressions.